zapcore/write_syncer.go

```diff
@@ -75,6 +78,87 @@ func (s *lockedWriteSyncer) Sync() error {
 	return err
 }
 
+type bufferWriterSyncer struct {
+	ws           WriteSyncer
+	bufferWriter *bufio.Writer
+}
+
+// defaultBufferSize sizes the buffer associated with each WriterSync.
+const defaultBufferSize = 256 * 1024
+
+// defaultFlushInterval means the default flush interval
+const defaultFlushInterval = 30 * time.Second
+
+// CancelFunc should be called when the caller exits to clean up buffers.
+type CancelFunc func() error
+
+// Buffer wraps a WriteSyncer in a buffer to improve performance,
+// if bufferSize = 0, we set it to defaultBufferSize
+// if flushInterval = 0, we set it to defaultFlushInterval
+func Buffer(ws WriteSyncer, bufferSize int, flushInterval time.Duration) (WriteSyncer, CancelFunc) {
```

Comment:
instead of `CancelFunc`, `io.Closer` may be a better interface,
https://golang.org/pkg/io/#Closer

Reply:
Maybe I should rename this func to CloseFunc? What do you think :-)